airflow/providers/amazon/aws/transfers/gcs_to_s3.py

```diff
@@ -164,14 +165,18 @@ def execute(self, context: 'Context') -> List[str]:
         if files:
 
             for file in files:
-                file_bytes = hook.download(object_name=file, bucket_name=self.bucket)
-
-                dest_key = self.dest_s3_key + file
-                self.log.info("Saving file to %s", dest_key)
-
-                s3_hook.load_bytes(
-                    file_bytes, key=dest_key, replace=self.replace, acl_policy=self.s3_acl_policy
-                )
+                with NamedTemporaryFile() as local_tmp_file:
+                    hook.download(object_name=file, bucket_name=self.bucket, filename=local_tmp_file.name)
```

Comment:
What do you think about `GCSHook.provide_file`?

Reply:
Changed the code to use `GCSHook.provide_file`. I didn't use it at first because in other operators i've seen `download_file` was used more frequently